Admin/Client/src/pages/loginPage.tsx

```diff
@@ -0,0 +1,101 @@
+    const [username, setUsername] = useState("");
+    const [password, setPassword] = useState("");
+    const [displayError, setError] = useState("");
+    const [isLoading, setLoading] = useState(false);
+    const [validated, setValidated] = useState(false);
+
+    const onLoginStart = () => {
+        setError("");
+        setLoading(true);
+    }
+
+    const onLoginFail = (error: string) => {
+        setError(error);
+    }
+
+    const onLoginSucces = () => {
+        setLoading(false);
+    };
+
+    const performLogin = (event: React.FormEvent<HTMLFormElement>) => {
+        const form = event.currentTarget;
+        event.preventDefault();
+        event.stopPropagation();
+        if (form.checkValidity() === false) {
+            setValidated(true);
+        }
+        else {
+            props.login(username,
+                password,
+                onLoginStart,
+                onLoginFail,
+                onLoginSucces);
+        }
+    }
+
+    return (
+        props.isAuthenticated ? <Redirect to="/" /> :
+            <LoginPageWrapper>
+                <LoginPageTitle>Login</LoginPageTitle>
+                <img src={logo} /><br />
+                {displayError ? <Alert variant="danger">
+                    <Alert.Heading>Oh snap!</Alert.Heading>
+                    {displayError}
+                </Alert> : false}
+                <Form noValidate validated={validated} onSubmit={performLogin}>
+                    <FormWrapper>
+                        <Form.Group controlId="formUsername">
+                            <Form.Label>Username</Form.Label>
+                            <Form.Control required type="text" placeholder="Username" value={username} onChange={e => setUsername(e.target.value)} />
+                            <Form.Control.Feedback type="invalid">
+                                Please enter a username.
+                        </Form.Control.Feedback>
+                        </Form.Group>
+                        <Form.Group controlId="formPassword">
+                            <Form.Label>Password</Form.Label>
+                            <Form.Control required type="password" placeholder="Password" value={password} onChange={e => setPassword(e.target.value)} />
+                            <Form.Control.Feedback type="invalid">
+                                Please enter a password.
+                        </Form.Control.Feedback>
+                        </Form.Group>
+                    </FormWrapper>
+                    <Button disabled={isLoading} variant="primary" type="submit">
+                        {isLoading ? 'Loading...' : 'Sign in'}
+                    </Button>
+                </Form>
+            </LoginPageWrapper>
+    )
+}
+
+const LoginPageWrapper = styled.div`
+    margin: auto;
+    width: 450px;
+    padding: 10px;
+    border: 1px solid #ededed;
+    border-radius: 2px;
+    text-align: center;
+    margin-top:50px;
+`
+
+const LoginPageTitle = styled.h1`
```

Comment:
if there is not css maybe it could be just h1?

Reply:
I will add some CSS so it's like a place holder